wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/monitoring/ClusterTopologyMonitorImpl.java

```diff
@@ -0,0 +1,851 @@
+              for (HostSpec hostSpec : hosts) {
+                this.nodeThreads.computeIfAbsent(hostSpec.getHost(),
+                    (key) -> {
+                      final Thread thread = this.getNodeMonitoringThread(hostSpec, this.writerHostSpec.get());
+                      thread.start();
+                      return thread;
+                    });
+              }
+            }
+            // otherwise let's try it again the next round
+
+          } else {
+
+            // node threads are running
+            // check if writer is already detected
+            final Connection writerConnection = this.nodeThreadsWriterConnection.get();
+            final HostSpec writerConnectionHostSpec = this.nodeThreadsWriterHostSpec.get();
+            if (writerConnection != null && writerConnectionHostSpec != null) {
+
+              this.closeConnection(this.monitoringConnection.get());
+              this.monitoringConnection.set(writerConnection);
+              this.writerHostSpec.set(writerConnectionHostSpec);
+              this.isVerifiedWriterConnection = true;
+              this.highRefreshRateEndTimeNano = System.nanoTime() + highRefreshRateAfterPanicNano;
+              this.ignoreNewFailoverRequestsEndTimeNano.set(System.nanoTime() + ignoreFailoverRequestNano);
+
+              this.nodeThreadsStop.set(true);
+              for (Thread thread : this.nodeThreads.values()) {
+                thread.interrupt();
+              }
+              this.nodeThreads.clear();
+
+              continue;
+
+            } else {
+              // update node threads with new nodes in the topology
+              List<HostSpec> hosts = this.nodeThreadsLatestTopology.get();
+              if (hosts != null && !this.nodeThreadsStop.get()) {
+                for (HostSpec hostSpec : hosts) {
+                  this.nodeThreads.computeIfAbsent(hostSpec.getHost(),
+                      (key) -> {
+                        final Thread thread = this.getNodeMonitoringThread(hostSpec, this.writerHostSpec.get());
+                        thread.start();
+                        return thread;
+                      });
+                }
+              }
+            }
+          }
+
+          this.delay(true);
+
+        } else {
+          // regular mode (not panic mode)
+
+          if (!this.nodeThreads.isEmpty()) {
+            // stop node threads
+            for (Thread thread : this.nodeThreads.values()) {
+              thread.interrupt();
+            }
+            this.nodeThreads.clear();
+          }
+
+          final List<HostSpec> hosts = this.fetchTopologyAndUpdateCache(this.monitoringConnection.get());
+          if (hosts == null) {
+            // can't get topology
+            // let's switch to panic mode
+            Connection conn = this.monitoringConnection.get();
+            this.monitoringConnection.set(null);
+            this.isVerifiedWriterConnection = false;
+            this.closeConnection(conn);
```

Comment:
Like to add a `continue` statement here?

Reply:
Yeah. My thinking was, if we are switching to panic mode we know there is likely a problem and we should start trying to fix it immediately instead of delaying below on line 382.